File: AppointmentSchedulerAPI/layers/BusinessLogicLayer/Model/GeneralNotification.cs
using AppointmentSchedulerAPI.layers.BusinessLogicLayer.Model.Types.Notification;

namespace AppointmentSchedulerAPI.layers.BusinessLogicLayer.Model
{
    public class GeneralNotification : Notification
    {
        public GeneralNotification()
        {
            this.Type = NotificationType.GENERAL_NOTIFICATION;
        }

        public required GeneralNotificationCodeType Code { get; set; }
    }
}

File: AppointmentSchedulerWeb/src/view-model/business-entities/notification/general-notification.ts
import { GeneralNotificationCodeType } from "../types/general-notification-code.types";
import { NotificationStatusType } from "../types/notification-status.types";
import { NotificationType } from "../types/notification.types";
import { NotificationBase } from "./notification-base";


export class GeneralNotification extends NotificationBase {
  constructor(
    createdAt: Date,
    uuid: string,
    message: string,
    type: NotificationType,
    status: NotificationStatusType,
    public code: GeneralNotificationCodeType) {
    super(createdAt, uuid, message, type, status);
  }

}
